apps/arena/lib/arena/configuration.ex

```diff
@@ -51,6 +54,16 @@ defmodule Arena.Configuration do
     Jason.decode!(payload.body, [{:keys, :atoms}])
   end
 
+  def get_consumable_items_configuration() do
+    gateway_url = Application.get_env(:arena, :gateway_url)
+
+    {:ok, payload} =
+      Finch.build(:get, "#{gateway_url}/curse/configuration/consumable_items", [{"content-type", "application/json"}])
+      |> Finch.request(Arena.Finch)
+
+    Jason.decode!(payload.body, [{:keys, :atoms}])
```

Comment:
Howdy!
I was looking at the PR and saw that this line is creating atoms dynamically (`{:keys, :atoms}` is basically a call to `String.to_atom`), which can led to a runtime crash (more info [here](https://curiosum.com/blog/elixir-anti-patterns#dynamic-atom-creation)). It is (almost) always recommended to use `String.to_existing_atom` and only allowing whitelisted atoms only.
In this case, you can change the decoding options to `[{:keys, :atoms!}]` which uses `String.to_existing_atom` under the hood, as [explained here](https://hexdocs.pm/jason/Jason.html#decode/2-options).

Feel free to reach me out.

Reply:
Thanks for the heads up! done in [043d63c](https://github.com/lambdaclass/mirra_backend/pull/742/commits/043d63cb653fab4190d038182b108ef9564cf5ca)